x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/engines_status/hooks/use_columns.tsx

```diff
@@ -0,0 +1,184 @@
+      },
+      {
+        field: 'id',
+        name: (
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.entityStore.enginesStatus.idColumnTitle"
+            defaultMessage="Identifier"
+          />
+        ),
+        render: (id: EngineComponentStatus['id'], { resource, installed }) => {
+          const path = getResourcePath(id, resource);
+
+          if (!installed || !path) {
+            return id;
+          }
+
+          return (
+            <EuiLink
+              target="_blank"
+              href={getUrlForApp('management', {
+                path,
+              })}
+            >
+              {id}
+            </EuiLink>
+          );
+        },
+      },
+      {
+        field: 'installed',
+        name: (
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.entityStore.enginesStatus.installedColumnTitle"
+            defaultMessage="Installed"
+          />
+        ),
+        width: '10%',
+        align: 'center',
+        render: (value: boolean) =>
+          value ? (
+            <EuiIcon data-test-subj="installation-status" type="check" color="success" size="l" />
+          ) : (
+            <EuiIcon data-test-subj="installation-status" type="cross" color="danger" size="l" />
+          ),
+      },
+      {
+        name: (
+          <FormattedMessage
+            id="xpack.securitySolution.entityAnalytics.entityStore.enginesStatus.healthColumnTitle"
+            defaultMessage="Health"
+          />
+        ),
+        width: '10%',
+        align: 'center',
+        render: ({ installed, resource, health }: EngineComponentStatus) => {
+          if (!installed) {
+            return null;
+          }
+
+          return <EuiHealth color={HEALTH_COLOR[health ?? 'green']} />;
+        },
+      },
+      {
+        isExpander: true,
+        align: 'right',
+        width: '40px',
+        name: (
+          <EuiScreenReaderOnly>
+            <span>
+              <FormattedMessage
+                id="xpack.securitySolution.entityAnalytics.entityStore.enginesStatus.expandRow"
+                defaultMessage="Expand row"
+              />
+            </span>
+          </EuiScreenReaderOnly>
+        ),
+        mobileOptions: { header: false },
+        render: (component: EngineComponentStatus) => {
+          const isItemExpanded = expandedItems.includes(component);
+
+          return component.errors && component.errors.length > 0 ? (
+            <EuiButtonIcon
+              onClick={() => onToggleExpandedItem(component)}
+              aria-label={isItemExpanded ? 'Collapse' : 'Expand'}
+              iconType={isItemExpanded ? 'arrowDown' : 'arrowRight'}
+            />
+          ) : null;
+        },
+      },
+    ],
+    [expandedItems, getUrlForApp, onToggleExpandedItem]
+  );
+};
+
+const getResourcePath = (id: string, resource: EngineComponentResource) => {
```

Comment:
Should I use a template and store it in a Map? 🤔

Reply:
i think that's slightly better yeah, but Id be okay with merging as is